File: Tasks/Domain/Tags/Tag.cs

namespace Domain.Tags;

public class Tag
{
    public Guid Id { get; }
    public String Name { get; }
    public String Color { get; }
    public Tag(Guid id, String name, String color)
    {
        Id = id;
        Name = name;
        Color = color;
    }
}

File: ClientApp/src/Domain/Tags/Models/Tag.ts
export class Tag {
  constructor(
    public readonly id: string,
    public readonly name: string,
    public readonly color: string
  ) {

  }
}

export function mapToTag(data: any): Tag {
  return new Tag(data.id, data.name, data.color)
}

export function mapToTags(data: any[]): Tag[] {
  return data.map(data => mapToTag(data))
}